### PR TITLE
Fix login credential type identifier parsing

### PR DESCRIPTION
**Pull request by ksamoray**
Values used for type identifier parsing were incorrect

### CODE REVIEW

File: nsxt/resource_nsxt_compute_manager.go [hunk 1]
```diff
@@ -449,7 +449,7 @@ func setCredentialValuesInSchema(d *schema.ResourceData, credential *data.Struct
 		return errors.New("no valid credential found")
 	}
 
-	d.Set("credential", elem)
+	d.Set("credential", []interface{}{elem})
```
Comment (salv-orlando) [inline nsxt/resource_nsxt_compute_manager.go:452]: was this wrong before and we did not realize only because elem was an empty map?

Comment (ksamoray) [reply nsxt/resource_nsxt_compute_manager.go:452]: Indeed. And the list was empty as the constants which were used in the switch case were wrong as well. So values were taken from the state.

File: nsxt/resource_nsxt_compute_manager.go [hunk 2]
```diff
@@ -435,7 +435,7 @@ func setCredentialValuesInSchema(d *schema.ResourceData, credential *data.Struct
 		elem["thumbprint"] = credEntry.Thumbprint
 		elem["password"] = credEntry.Password
```
Comment (annakhm) [inline nsxt/resource_nsxt_compute_manager.go:436]: Would we see non-empty diff here since sensitive data is not sent back by NSX?

Comment (ksamoray) [reply nsxt/resource_nsxt_compute_manager.go:436]: Yep - in transport node the polymorphic objects are recreated with make(map[]interface{}), while here the same map is being populated with the GET values. So passwords are retained from the state.
I'm implementing a similar mechanism for transport node.

Comment (annakhm) [reply nsxt/resource_nsxt_compute_manager.go:436]: Sorry - I'm still missing something.. I see `elem` is initialized as an empty map, and then populated with `GET` values. I'm missing the part where passwords are coming from the state..

**Review by annakhm — APPROVED** [review]

**Pull request merged**